src/xlat.c

```diff
@@ -602,18 +602,15 @@ enum xlat_mode xlat_get_mode(void)
 
 void xlat_auto_trigger_action(void)
 {
-    // random delay, such that we do not always perfectly align with USB timing
-    srand(xTaskGetTickCount());
-    int val = rand() & 0xFFF;
-    for (volatile int i = 0; i < val; i++) {
-        __NOP();
-    }
-    HAL_GPIO_WritePin(ARDUINO_D11_GPIO_Port, ARDUINO_D11_Pin, auto_trigger_level_high ? GPIO_PIN_SET : GPIO_PIN_RESET);
+    uint32_t delay = rand() % 1000 + 100;
+    last_btn_gpio_timestamp = xlat_counter_1mhz_get() + delay;
+    htim2.Instance->CCR1 = last_btn_gpio_timestamp;
```

Comment:
I think there's a potential issue if we get interrupted for longer than the chosen delay time before this line as the output compare function checks for exact match, so we might miss it. Could possibly solve it with PWM mode or just making the delay long enough.